File: elixir/test/day01_test.exs
defmodule Day01Test do
  use ExUnit.Case, async: true
  import Advent.Day01

  @moduletag :day01

  @example1 """
  1000
  2000
  3000

  4000

  5000
  6000

  7000
  8000
  9000

  10000
  """

  setup_all do
    {:ok, data} = File.read("/data/day01.txt")
    {:ok, data: data}
  end

  describe "examples" do
    @tag examples: :part1
    test "part1" do
      assert part1(@example1) == 24000
    end

    @tag examples: :part2
    test "part2" do
      assert part2(@example1) == 45000
    end
  end

  describe "solutions" do
    @tag solutions: :part1
    test "part1", state do
      assert part1(state[:data]) == 68775
    end

    @tag solutions: :part2
    test "part2", state do
      assert part2(state[:data]) == 202585
    end
  end
end

File: clojure/test/day01_test.clj
(ns day01-test
  (:require [clojure.test :refer [deftest is testing use-fixtures]]
            day01))

;; https://stuartsierra.com/2016/05/19/fixtures-as-caches
(def ^:dynamic data)
(defn load-data [f]
  (binding [data (slurp "/data/day01.txt")] 
    (f)))
(use-fixtures :each load-data)

(def EXAMPLE1 (clojure.string/trim "
1000
2000
3000

4000

5000
6000

7000
8000
9000

10000
"))

(deftest day01-examples-test
  (is (and
    (= 24000 (day01/part1 EXAMPLE1))
    (= 45000 (day01/part2 EXAMPLE1)))))

(deftest day01-part1-test
  (is (= 68775 (day01/part1 data))))

(deftest day01-part2-test
  (is (= 202585 (day01/part2 data))))